tests/optimize/test_spsa.py

```diff
@@ -442,6 +442,44 @@ def cost(params):
         ):
             opt.step(cost, params)
 
+    @pytest.mark.usefixtures("use_legacy_opmath")
+    @pytest.mark.slow
+    def test_lighting_device(self):
```

Comment:
I thought legacy test cases should be post-fixed with `_legacy`?

Reply:
Indeed, will change 👍 `_legacy_opmath()` in fact

Reply:
```suggestion
    def test_lighting_device_legacy_opmath(self):
```